lib/logger.js

```diff
@@ -13,12 +13,16 @@ var os = require("os"),
         formatversion: cfg.versions[type],
         type: cfg.servicetype + "-" + type + "-" + cfg.versions[type],
         host: cfg.host,
-        sequencenumber: cfg.sequencenumber,
+        sequencenumber: cfg.sequencenumber
       };
 
       return hoek.applyToDefaults(fields, log);
     },
     _dispatch = function(log){
+      if (log.headers && cfg.validate && !cfg.validate(log)) {
```

Comment:
Do we care about checking headers here? I think you're using the `log.headers` param to say "is this a request log" in which case, the `validate()` method will only get called for request logs, instead of all logs. If this is the intended behaviour then this check should live inside the `request()` method (up a level) so that it's clear.

The alternative would be to remove the `log.headers` check and thereby allow you to scan (and possibly reject) all types of logs (whether they are `log`, `error` or `request`)